lib/utils/resolveFilePath.mjs

```diff
@@ -0,0 +1,45 @@
+import { join, normalize, resolve, sep } from 'node:path';
+import { lstatSync } from 'node:fs';
+
+/**
+ * Return the resolved file path, based on whether the provided parameter is
+ * a directory or looks like a directory (ends in `path.sep`), in which case the file
+ * name will be `location/defaultFileName`.
+ *
+ * If location points to a file or looks like a file, then it will just use that file.
+ *
+ * @param {string} location - The name of file or directory to be used
+ * @param {string} cwd - Current working directory. Used for tests
+ * @param {string} defaultFileName - Default filename to use when location is a directory
+ * @returns {string} Resolved path to the file
+ */
+export default function resolveFilePath(location, cwd, defaultFileName) {
```

Comment:
[suggestion] If I understand correctly (sorry if I missed some comments), `resolveFilePath()` seems to be introduced to replace the existing `getCacheFile()`. This replacement is surely reasonable, but it seems to be out of the scope of this PR, which adds a new experimental feature.

Since `getCacheFile()` is quite old code, I wouldn't like to mix refactoring and a new feature in one PR. I worry that `git blame` might be difficult later.

So, I'd appreciate it if you could extract the refactoring into another PR.

Reply:
Thanks, I've merged the refactoring PR. 👍🏼